// Java: Exception
// JavaScript: Error
// const array = new Array(100000000000000000);

// Error(Exception) Handling: try -> catch -> finally

function readFile(fileName: string): string {
  if (fileName === 'not exist!ðŸ’©') {
    throw new Error(`file not exist! ${fileName}`);
  }
  return 'file contentsðŸ—’';
}

function closeFile(fileName: string) {
  //
}
function run() {
  const fileName = 'not exist!ðŸ’©';

  try {
    console.log(readFile(fileName));
  } catch (error) {
    console.log(`catched!!`);
    return;
  } finally {
    closeFile(fileName);
    console.log(`closed!`);
  }
}
run();
